eth/backend.go

```diff
@@ -87,12 +90,23 @@ func (b *backend) SendTransaction(ctx context.Context, tx *types.Transaction) er
 		method = "unknown"
 	}
 
+	txParams := make(map[string]interface{})
+	err = decodeTxParams(b.abiMap[string(data[:4])], txParams, data)
 	if err != nil {
-		glog.Infof("\n%vEth Transaction%v\n\nInvoking transaction: \"%v\".  \nTransaction Failed: %v\n\n%v\n", strings.Repeat("*", 30), strings.Repeat("*", 30), method, err, strings.Repeat("*", 75))
 		return err
 	}
 
-	glog.Infof("\n%vEth Transaction%v\n\nInvoking transaction: \"%v\".  Hash: \"%v\". \n\n%v\n", strings.Repeat("*", 30), strings.Repeat("*", 30), method, tx.Hash().String(), strings.Repeat("*", 75))
+	var txParamsString string
+	for name, val := range txParams {
+		txParamsString += fmt.Sprintf("%v: %v   ", name, val)
+	}
+
+	if err != nil {
```

Comment:
We dont need this `nil` check here.  We tested the return value of `decodeTxParams()` just above, and there hasn't been any reassignment to `err` since.  `err` cant be `nil` here because we would have already returned with `err`

Reply:
The original intention here is to display the transaction failed `glog` statement but this codepath can never be reached as you say which was already the case before this PR. 

Therefore I have renamed the error returned from `SendTransaction` and we delay the check until we have decoded the tx params. d3504b70